### PR TITLE
macmon: init at v0.3.3

### PR DESCRIPTION
**Pull request by schrobingus**
Macmon is a sudoless performance monitor for Apple Silicon chips, built using Rust for macOS.

https://github.com/vladkens/macmon

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/os-specific/darwin/macmon/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  darwin,
```
Comment (matteo-pacini) [inline pkgs/os-specific/darwin/macmon/default.nix:5]:
```suggestion
```

File: pkgs/os-specific/darwin/macmon/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  src = fetchFromGitHub {
+    owner = "vladkens";
+    repo = "macmon";
+    rev = "v${version}";
+    hash = "sha256-COpEjK1LUwGzhSgD09D4gx+MtS2hT0qt06rTPT8JQiQ=";
+  };
+
+  cargoHash = "sha256-DTkpFGl8kTWttFGKTCpny2L0IRrCgpnnXaKIFoxWrW4=";
+
+  buildInputs = [ darwin.apple_sdk.frameworks.IOKit ];
```
Comment (matteo-pacini) [inline pkgs/os-specific/darwin/macmon/default.nix:21]:
```suggestion
```

Not needed anymore, as `apple-sdk` is implicitly added to the `stdenv` when on Darwin (all frameworks are directly available).

File: pkgs/os-specific/darwin/macmon/default.nix (hunk 1 above)
Comment (matteo-pacini) [inline pkgs/os-specific/darwin/macmon/default.nix:5]:
```suggestion
```

File: pkgs/os-specific/darwin/macmon/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/vladkens/macmon";
+    description = "Sudoless performance monitoring for Apple Silicon processors";
+    platforms = platforms.darwin;
+    license = licenses.mit;
+    maintainers = with maintainers; [ schrobingus ];
+  };
+}
```
Comment (matteo-pacini) [inline pkgs/os-specific/darwin/macmon/default.nix:30]:
```suggestion
  meta = {
    homepage = "https://github.com/vladkens/macmon";
    description = "Sudoless performance monitoring for Apple Silicon processors";
    platforms = lib.platforms.darwin;
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ schrobingus ];
  };
}
```
As per #208242

**Review by matteo-pacini — CHANGES_REQUESTED** [review]
Thank you for your contribution, left a few comments

**Comment by schrobingus** [comment]
@matteo-pacini Changes applied :^)

**Comment by schrobingus** [comment]
Good point, I'll move the definition over there.

For reference, this tool is meant to be ran on aarch64 Macs in particular since it's a monitor specifically for Apple Silicon chips. Should I dedicate the architecture in the package, or is it fine as is?

If the former, would I do it like this under `platform`?
```nix
arch = "arm64";
```

**Comment by schrobingus** [comment]
Should be good to go!

**Review by matteo-pacini — APPROVED** [review]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 354348`

---
### `x86_64-darwin`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>macmon</li>
  </ul>
</details>

LGTM

**Pull request merged**